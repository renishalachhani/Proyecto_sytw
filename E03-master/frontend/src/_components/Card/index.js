import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import { assets } from '../../_helpers/assets'

const useStyles = makeStyles({
  card: {
    maxWidth: 345,
    alignItems: 'center',
    margin: '0 20px',
  },
})
export default function ImgMediaCard(props) {
  const classes = useStyles()
  const { title, src, fechaInicio, descripcion, localizacion } = props
  return (
    <Card className={classes.card} onClick={props.onClick}>
      <CardActionArea>
        <CardMedia
          component="img"
          alt={title}
          height="140"
          image={assets('/images/book.jpg')}
          title={title}
        />
        <CardContent>
          <Typography className="text-truncate" style={{ maxWidth: "230px" }} gutterBottom variant="h5" component="p">
            {title}
          </Typography>
          <Typography className="text-truncate" style={{ maxWidth: "240px" }} variant="body2" color="textSecondary" component="p">
            {descripcion}
          </Typography>
          <br />
            <div className="text-truncate" style={{ maxWidth: "240px" }}>
              Place: {localizacion}
            </div>
          <br /> <br />
          Date: {fechaInicio}
        </CardContent>
      </CardActionArea>
    </Card>
  )
}
