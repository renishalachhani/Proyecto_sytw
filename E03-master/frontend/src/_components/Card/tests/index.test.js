import jsdom from 'jsdom'
import React from 'react'
import { mount, configure } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'

import Card from '../index'

configure({ adapter: new Adapter() })

const { JSDOM } = jsdom
const dom = new JSDOM('<!doctype html><html><body></body></html>')
const { window } = dom

global.window = window
global.document = window.document

describe('<Card />', () => {
  it('Comprobar que se renderiza Card', () => {
    const wrapper = mount(
        <Card
          src="/images/book.jpg"
          title="Firma de Libros de Miguel Noguera"
          fecha="8 de Septiembre"
          desc="El autor estará firmando libros en la libreria Lemus de La Laguna"
          id="1"
        />
    )
    expect(wrapper.find(Card)).toHaveLength(1)
  })

  it('Snapshot', () => {  
    const Component = renderer.create(<Card 
      src="/images/book.jpg"
      title="Firma de Libros de Miguel Noguera"
      fecha="8 de Septiembre"
      desc="El autor estará firmando libros en la libreria Lemus de La Laguna"
      id="1"
    />).toJSON();
    expect(Component).toMatchSnapshot();
  });
})
