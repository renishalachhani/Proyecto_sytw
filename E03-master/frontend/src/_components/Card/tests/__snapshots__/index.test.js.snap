// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="Firma de Libros de Miguel Noguera"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="140"
      src="http://localhost:4000/images/book.jpg"
      title="Firma de Libros de Miguel Noguera"
    />
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root text-truncate MuiTypography-h5 MuiTypography-gutterBottom"
        style={
          Object {
            "maxWidth": "230px",
          }
        }
      >
        Firma de Libros de Miguel Noguera
      </p>
      <p
        className="MuiTypography-root text-truncate MuiTypography-body2 MuiTypography-colorTextSecondary"
        style={
          Object {
            "maxWidth": "240px",
          }
        }
      />
      <br />
      <div
        className="text-truncate"
        style={
          Object {
            "maxWidth": "240px",
          }
        }
      >
        Place: 
      </div>
      <br />
       
      <br />
      Date: 
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
